tracing-core/src/metadata.rs

```diff
@@ -57,6 +46,7 @@ use core::{
 /// [module path]: Self::module_path
 /// [collector]: super::collect::Collect
 /// [callsite identifier]: super::callsite::Identifier
+#[derive(Eq, PartialEq)]
```

Comment:
we may want a hand-written equality implementation for `Metadata`s so that we can compare the callsite identifier first, and short-circuit so that we don't have to do a bunch of string equality. i believe the derived `PartialEq` impl will compare fields in the order that they are declared.

that change could be made later as an optimization, though.

Reply:
actually, i believe the derived implementations also generate an implementation of the [`StructuralPartialEq` trait](https://doc.rust-lang.org/std/marker/trait.StructuralPartialEq.html), which allows matching a type as a match pattern (which is possible in v0.1.x of `tracing-core`, where `Metadata`'s fields are `pub`). this means that a change from a hand-written implementation to a derived one is potentially a breaking change...